### ISSUE TITLE
Warn about transposed characters in variable length expansion

### ISSUE THREAD

**Issue by sorpigal** [koalaman/shellcheck#2924]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
n=0
((n++))
: "$#{n}"
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

```
Line 4:
: "$#{n}"
    ^-- [SC????](https://www.shellcheck.net/wiki/SC???? (warning): $#{ probably was intended to be ${# here
```

**Comment by koalaman** [comment]
You're right, of course. I looked through a corpus of shell scripts but did not find a significant number of instances to warrant a specific check for this particular mistake. Thanks for the suggestion!
